etc/tests/test_bootstrapper.py

```diff
@@ -0,0 +1,289 @@
+
+    bootstrapper.put_file_to_object_storage(s3_setup, bucket_name, file_to_put)
+
+    with tmpdir.as_cwd():
+        s3_setup.get_object(bucket_name, file_to_put)
+
+
+def test_fail_invalid_filename_put_file_object_store(monkeypatch, s3_setup):
+    bucket_name = "test-bucket"
+    file_to_put = "LICENSE_NOT_HERE"
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", lambda x: file_to_put)
+
+    with pytest.raises(FileNotFoundError):
+        bootstrapper.put_file_to_object_storage(s3_setup, bucket_name, file_to_put)
+
+
+def test_fail_bucket_put_file_object_store(monkeypatch, s3_setup):
+    bucket_name = "test-bucket-not-exist"
+    file_to_put = "LICENSE"
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", lambda x: file_to_put)
+
+    with pytest.raises(minio.error.NoSuchBucket):
+        bootstrapper.put_file_to_object_storage(s3_setup, bucket_name, file_to_put)
+
+
+def test_parse_arguments():
+    test_args = ['-e', 'http://test.me.now',
+                 '-d', 'test-directory',
+                 '-t', 'test-archive.tgz',
+                 '-i', 'test-notebook.ipynb',
+                 '-b', 'test-bucket']
+    bootstrapper.parse_arguments(test_args)
+
+
+def test_fail_missing_notebook_parse_arguments():
+    test_args = ['-e', 'http://test.me.now',
+                 '-d', 'test-directory',
+                 '-t', 'test-archive.tgz',
+                 '-b', 'test-bucket']
+    with pytest.raises(SystemExit):
+        bootstrapper.parse_arguments(test_args)
+
+
+def test_fail_missing_endpoint_parse_arguments():
+    test_args = ['-d', 'test-directory',
+                 '-t', 'test-archive.tgz',
+                 '-i', 'test-notebook.ipynb',
+                 '-b', 'test-bucket']
+    with pytest.raises(SystemExit):
+        bootstrapper.parse_arguments(test_args)
+
+
+def test_fail_missing_archive_parse_arguments():
+    test_args = ['-e', 'http://test.me.now',
+                 '-d', 'test-directory',
+                 '-i', 'test-notebook.ipynb',
+                 '-b', 'test-bucket']
+    with pytest.raises(SystemExit):
+        bootstrapper.parse_arguments(test_args)
+
+
+def test_fail_missing_bucket_parse_arguments():
+    test_args = ['-e', 'http://test.me.now',
+                 '-d', 'test-directory',
+                 '-t', 'test-archive.tgz',
+                 '-i', 'test-notebook.ipynb']
+    with pytest.raises(SystemExit):
+        bootstrapper.parse_arguments(test_args)
+
+
+def test_fail_missing_directory_parse_arguments():
+    test_args = ['-e', 'http://test.me.now',
+                 '-t', 'test-archive.tgz',
+                 '-i', 'test-notebook.ipynb',
+                 '-b', 'test-bucket']
+    with pytest.raises(SystemExit):
+        bootstrapper.parse_arguments(test_args)
+
+
+def test_requirements_file():
+    requirements_file = "etc/tests/resources/test-requirements-elyra.txt"
+    correct_number_of_packages = 18
```

Comment:
We might want to calculate this although the test will break if the requirements were to change, so calculation may be more of a hassle than its worth.

Reply:
skipping the test for now, will leave as informational in case I think of a better way to calculate automatically

Reply:
I think the test is fine - no need to skip.  Just saying we should try to avoid hard-coded counts like this in general.